generators/add-licence-headers/index.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+const fs = require('fs');
+const Base = require('yeoman-generator').Base;
+const chalk = require('chalk');
+const glob = require('glob');
+
+const defaultLicenceFileName = 'licence.txt';
+const defaultLicencePath = `./${defaultLicenceFileName}`;
+
+const comments = {
+  ts: {
+    begin: '/*',
+    body: '*',
+    end: '*/',
+    aligningSpaces: 1
+  },
+  py: {
+    begin: '##',
+    body: '#',
+    end: '##',
+    aligningSpaces: 0
+  },
+  scss: {
+    begin: '/*',
+    body: '*',
+    end: '*/',
+    aligningSpaces: 1
+  }
+};
+
+class Generator extends Base {
+  constructor(args, options) {
+    super(args, options);
+
+    this.comments = {};
+
+    this.option('licencePath', {
+      alias: 'l',
+      default: defaultLicencePath,
+      type: String,
+      desc: `Relative path to a ${chalk.blue(defaultLicenceFileName)} file`
+    });
+
+    this.option('plugins', {
+      alias: 'p',
+      default: '',
+      type: String,
+      desc: 'Comma separated list (without spaces) of plugins to prepend the licence to'
+    });
+
+    this.option('excludedFileTypes', {
+      alias: 'e',
+      default: '',
+      type: String,
+      desc: 'Comma separated list (without spaces) to exclude specific file types (e.g. only add to .ts files by excluding .scss and .py)'
+    });
+  }
+
+  prompting() {
+    return this.prompt([
+      {
+        type: 'input',
+        name: 'licencePath',
+        message: `Please enter the relative path to the ${chalk.blue(defaultLicenceFileName)} file`,
+        when: this.options.licencePath === defaultLicencePath
+      },
+      {
+        type: 'checkbox',
+        name: 'excludedFileTypes',
+        message: 'Exclude file types from adding headers',
+        choices: Object.keys(comments),
+        when: this.options.excludedFileTypes.length === 0
+      }
+    ]).then((props) => {
+      this.licencePath = props.licencePath || this.options.licencePath;
+      const excludedFileTypes = props.excludedFileTypes || this.options.excludedFileTypes.split(',');
+
+      // filter out excluded file types
+      this.fileTypes = Object.keys(comments).filter((type) => !excludedFileTypes.includes(type));
+
+      // TODO: How to abort the generator correctly?
+      return this._readLicenceFile();
+    }).catch((err) => this.log(err));
+  }
+
+  writing() {
+    this._generateComments();
+
+    try {
+      const sourceFolders = this._getSourceFolders();
+
+      sourceFolders.forEach((folderName) => {
+        const pluginPath = this.destinationPath(folderName);
```

Comment:
what has a plugin path to do with this variable?

Reply:
would `sourcePath` be a better name?